backend/src/controllers/userController.js

```diff
@@ -18,23 +18,34 @@ const getUser = (req, res) => {
 
 const createUser = (req, res, next) => {
   const { username, password, role } = req.body;
+
+  const requiredFields = ["username", "password", "role"];
+  const missingFields = requiredFields.filter((field) => !req.body[field]);
+
   const userExists = users.find((user) => user.username === username);
 
-  if (userExists) {
-    res.status(409).json({
-      status: "Fail",
-      message: "User already exists!",
+  if (missingFields.length > 0) {
+    res.status(422).json({
```

Comment:
Nice to use 422 but you can also use 400 which is mostly used to a bad request. Both are OK but mostly in real life people use more of 400